packages/oauth/src/tokenSession.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Juno contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { parseIdTokenData, decodeIDToken } from "./tokenHelpers"
+
+interface AuthData {
+  JWT: string
+  raw: Record<string, any>
+  refreshToken: string
+  parsed: any // Adjust this type as per the return type of parseIdTokenData
+}
+
+export const composeAuthData = (token: string, options: any): AuthData => {
+  let tokenData = {}
+  try {
+    tokenData = decodeIDToken(token)
+  } catch (_) {
+    console.warn(`WARNING: (OAUTH) Could not parse token: ${token}`)
+  }
+
+  // extend tokenData with the options
+  tokenData = { ...tokenData, ...options }
+
+  return {
+    JWT: token,
+    raw: tokenData,
+    refreshToken: "TOKEN",
+    parsed: parseIdTokenData(tokenData),
+  }
+}
+
+interface TokenSessionReturnType {
+  login: () => void
+  logout: () => void
+  currentState: () => Record<string, any>
+}
+
+export default function tokenSession(params: any = {}): TokenSessionReturnType {
```

Comment:
I do not think that you need to specify params as any, please provide a type accodring to params that you expect. at least token, onUpdate, options and then you can use key syntax of other object related params

Reply:
Yes, you are right. This is a copy paste error...

Reply:
fixed